src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileMetadataRepositorySpec.scala

```diff
@@ -0,0 +1,99 @@
+package uk.gov.nationalarchives.tdr.api.db.repository
+
+import java.util.UUID
+
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import uk.gov.nationalarchives.tdr.api.db.DbConnection
+import uk.gov.nationalarchives.tdr.api.utils.TestUtils
+import uk.gov.nationalarchives.tdr.api.utils.TestUtils._
+
+
+class FileMetadataRepositorySpec extends AnyFlatSpec with ScalaFutures with Matchers {
+
+  "countProcessedChecksumInConsignment" should "return 0 if a consignment has no files" in {
+    val db = DbConnection.db
+    val fileMetadataRepository = new FileMetadataRepository(db)
+    val consignmentId = UUID.fromString("c03fd4be-58c1-4cee-8d3c-d162bb4f7c01")
+
+    TestUtils.createConsignment(consignmentId, userId)
+
+    val consignmentFiles = fileMetadataRepository.countProcessedChecksumInConsignment(consignmentId).futureValue
+
+    consignmentFiles shouldBe 0
+  }
+
+  "countProcessedChecksumInConsignment" should "return 0 if consignment has no checksum metadata for files" in {
+    val db = DbConnection.db
+    val fileMetadataRepository = new FileMetadataRepository(db)
+    val consignmentId = UUID.fromString("64456c78-49bb-4bff-85c8-2fff053b9f8d")
+    val fileOneId = UUID.fromString("2c2dedf5-56a1-497f-9f8a-19102739a056")
+    val fileTwoId = UUID.fromString("c2d5c569-0fc4-4688-9523-157c4028f1b1")
+    val fileThreeId = UUID.fromString("317c7084-d3d4-435b-acd2-cfb317793844")
+
+    TestUtils.createConsignment(consignmentId, userId)
+    TestUtils.createFile(fileOneId, consignmentId)
+    TestUtils.createFile(fileTwoId, consignmentId)
+    TestUtils.createFile(fileThreeId, consignmentId)
+//  We have created files, but not added any fileMetadata for those files to the AVMetadata repository.
```

Comment:
Looks good! Should this refer to the FileProperty repository rather than the AVMetadata repository?

Reply:
Good spot! Thanks!